databricks/koalas/indexes.py

```diff
@@ -144,6 +144,36 @@ def to_pandas(self) -> pd.Index:
 
     toPandas = to_pandas
 
+    def to_numpy(self, dtype=None, copy=False):
+        """
+        A NumPy ndarray representing the values in this Index or MultiIndex.
+
+        .. note:: This method should only be used if the resulting NumPy ndarray is expected
+            to be small, as all the data is loaded into the driver's memory.
+
+        Parameters
+        ----------
+        dtype : str or numpy.dtype, optional
+            The dtype to pass to :meth:`numpy.asarray`
+        copy : bool, default False
+            Whether to ensure that the returned value is a not a view on
+            another array. Note that ``copy=False`` does not *ensure* that
+            ``to_numpy()`` is no-copy. Rather, ``copy=True`` ensure that
+            a copy is made, even if not strictly necessary.
+
+        Returns
+        -------
+        numpy.ndarray
+
+        Examples
+        --------
+        >>> ks.Series([1, 2, 3, 4]).index.to_numpy()
+        array([0, 1, 2, 3])
+        >>> ks.DataFrame({'a': ['a', 'b', 'c']}, index=[[1, 2, 3], [4, 5, 6]]).index.to_numpy()
+        array([(1, 4), (2, 5), (3, 6)], dtype=object)
+        """
+        return self.to_pandas().to_numpy(dtype=dtype, copy=copy)
```

Comment:
Seems like pandas 0.23 does not have this API. Can you check pandas's to_numpy and mimic it here?

If that's not possible or complicated, we should skip the doctests and document Koalas with pands 0.23 does not support this API.

Reply:
Thanks for pointing out! Yes pandas add this api since `0.24`. 

It is not complicated to mimic. But personally I think some part of code is not stable. I see some WIP PR in the comment. 
https://github.com/pandas-dev/pandas/blob/v0.25.3/pandas/core/base.py#L916-L1010
Currently if index.dtype is datetime and parameter dtype is none, it would set it as object.

If call pandas directly we could have consistent behaviors with pandas. 

If mimic the it, I would do 

```python
        result = np.asarray(self.to_pandas()._values, dtype=dtype)
        if copy:
            result = result.copy()
        return result
```


Any suggestion?

Reply:
Thanks for pointing it out. Hm, let's just copy as is for now. I'm periodically checking the existing APIs and maybe I would be able to sync it again when pandas happens to have some diff later here.